internal/datastore/kvstore.go

```diff
@@ -0,0 +1,170 @@
+package datastore
+
+import (
+	"context"
+
+	apiv1 "github.com/syncromatics/kvetch/internal/protos/kvetch/api/v1"
+
+	badger "github.com/dgraph-io/badger/v2"
+	"github.com/pkg/errors"
+)
+
+// KVStore is the key value datastore
+type KVStore struct {
+	db *badger.DB
+}
+
+// NewKVStore creates a new key value datastore
+func NewKVStore(path string) (*KVStore, error) {
+	db, err := badger.Open(badger.DefaultOptions(path))
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to open datastore")
+	}
+	return &KVStore{
+		db,
+	}, nil
+}
+
+// Get retrieves key values from the datastore.
+func (s *KVStore) Get(request *apiv1.GetValuesRequest) (*apiv1.GetValuesResponse, error) {
+	response := &apiv1.GetValuesResponse{
+		Messages: []*apiv1.KeyValue{},
+	}
+
+	err := s.db.View(func(txn *badger.Txn) error {
+		for _, key := range request.Requests {
+			if key.IsPrefix {
+				values, err := s.prefixScan(txn, key.Key)
+				if err != nil {
+					return errors.Wrap(err, "failed prefix scan")
+				}
+				response.Messages = append(response.Messages, values...)
+				continue
+			}
+			value, err := txn.Get([]byte(key.Key))
+			if err == badger.ErrKeyNotFound {
```

Comment:
any use in returning an extra object on the response listing keys that were not found?